### PR TITLE
Redesign of colors, fonts and text size

### PR DESCRIPTION
**Pull request by logotet**

### CODE REVIEW

File: app/src/main/java/com/example/weatherapp/models/utils/DataFormatter.kt [hunk 1]
```diff
@@ -39,7 +39,20 @@ fun Long.formatHour(resourceProvider: ResourceProvider, timeOffset: Int?): Strin
     timeOffset?.let {
         val zoneOffset = ZoneOffset.ofTotalSeconds(timeOffset)
         val dateTime = LocalDateTime.ofEpochSecond(this, 0, zoneOffset)
-        format = resourceProvider.getString(R.string.hour_format, dateTime.hour.toString())
```
Comment (sando22) [inline app/src/main/java/com/example/weatherapp/models/utils/DataFormatter.kt:42]: Check how to format date and time with `SimpleDateFormat`

File: app/src/main/java/com/example/weatherapp/models/utils/DataFormatter.kt [hunk 2]
```diff
@@ -39,7 +39,20 @@ fun Long.formatHour(resourceProvider: ResourceProvider, timeOffset: Int?): Strin
...
+            format = resourceProvider.getString(
+                R.string.hour_format_single_digit,
+                dateTime.hour.toString()
+            )
+        }
     }
     return format
+}
+
+fun String.formatDescription(): String {
```
Comment (sando22) [inline app/src/main/java/com/example/weatherapp/models/utils/DataFormatter.kt]: Think of better name as it is visible globally. Hint: what this function is doing

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -13,7 +13,8 @@
     //Measure format
     <string name="temperature_format">%.0f%s</string>
     <string name="speed_format">%.1f%s</string>
-    <string name="hour_format">%sh</string>
+    <string name="hour_format">%s:00</string>
+    <string name="hour_format_single_digit">0%s:00</string>
```
Comment (sando22) [inline app/src/main/res/values/strings.xml]: Probably won't need these

**Review by sando22 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/weatherapp/models/utils/DataFormatter.kt (hunk 1 above)
Comment (logotet) [reply app/src/main/java/com/example/weatherapp/models/utils/DataFormatter.kt:42]: Resolved with DateTimeFormatter

File: app/src/main/java/com/example/weatherapp/models/utils/DataFormatter.kt [hunk 4]
```diff
@@ -34,12 +35,19 @@ fun Double?.formatSpeed(resourceProvider: ResourceProvider, unitSystem: UnitSyst
 }
 
 @RequiresApi(Build.VERSION_CODES.O)
-fun Long.formatHour(resourceProvider: ResourceProvider, timeOffset: Int?): String {
+fun Long.formatHour(timeOffset: Int?): String {
     var format = ""
+    val formatter: DateTimeFormatter = DateTimeFormatter.ofPattern("HH:mm")
     timeOffset?.let {
         val zoneOffset = ZoneOffset.ofTotalSeconds(timeOffset)
```
Comment (sando22) [inline app/src/main/java/com/example/weatherapp/models/utils/DataFormatter.kt:42]: Have you tested this with null timeOffset?

File: app/src/main/java/com/example/weatherapp/ui/hours/HourViewModel.kt [hunk 5]
```diff
@@ -47,7 +47,7 @@ class HourViewModel @Inject constructor(
     @get:Bindable
     val hourValue: String?
         @RequiresApi(Build.VERSION_CODES.O)
-        get() = hourModel?.hour?.formatHour(resourceProvider, hourModel?.timeZoneOffset)
+        get() = hourModel?.timeZoneOffset?.let { hourModel?.hour?.formatHour(it) }
```
Comment (sando22) [inline app/src/main/java/com/example/weatherapp/ui/hours/HourViewModel.kt]: there is a better way of doing this check as `hourModel?.` is redundant

File: app/src/main/java/com/example/weatherapp/models/utils/DataFormatter.kt [hunk 6]
```diff
@@ -33,13 +34,30 @@ fun Double?.formatSpeed(resourceProvider: ResourceProvider, unitSystem: UnitSyst
...
+//    val formatter: DateTimeFormatter = DateTimeFormatter.ofPattern("HH:mm")
+//    timeOffset?.let {
+//        OffsetTime
+//        ZoneOffset.of
+//        val zoneOffset = ZoneOffset.ofTotalSeconds(timeOffset)
+//        val dateTime = LocalDateTime.ofEpochSecond(this, 0, zoneOffset)
+//        format = dateTime.format(formatter)
+//    }
+//    return format
+//}
```
Comment (sando22) [inline app/src/main/java/com/example/weatherapp/models/utils/DataFormatter.kt]: Did you leave that intentionally

File: app/src/main/java/com/example/weatherapp/ui/hours/HourViewModel.kt (hunk 5 above)
Comment (sando22) [reply app/src/main/java/com/example/weatherapp/ui/hours/HourViewModel.kt]: `hourModel` is nullable and you check if it is not null twice. Change it so the check is done only once .

**Review by sando22 — APPROVED** [review]

**Pull request merged**